src/lib.rs

```diff
@@ -20,16 +24,37 @@ pub struct Config {
     pub redirect_url: String,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+// See https://tools.ietf.org/html/rfc6749#section-5.1
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Deserialize)]
 pub struct Token {
     pub access_token: String,
+    #[serde(default)]
     pub scopes: Vec<String>,
     pub token_type: String,
+    #[serde(default)]
+    pub expires_in: Option<u32>,
+    #[serde(default)]
+    pub refresh_token: Option<String>,
+}
+
+// See https://tools.ietf.org/html/rfc6749#section-5.2
+#[derive(Debug, Deserialize)]
+struct TokenError {
```

Comment:
well actually it's a bit more complicated. on one side, if I make the `TokenError` public, it won't really do much unless I also get all those `exchange` methods return `Result<Token, TokenError>`. however, if I do the latter, I can't return the (json) parse errors any more, cause I'd have to fit them somehow in the `TokenError` pattern.

regarding the `TokenError`, the `error` field should/could be an `Enum` (the [RFC](https://tools.ietf.org/html/rfc6749#section-4.1.2.1) mentions a strict set of values) - but again this depends on some investigation that I'll have to do whether popular APIs (let's say Google, Github, Twitter) honour this.

so please go ahead with merging this, and I'll follow up with some more stuff (including tests) and some questions :)

personally I don't think that adding extra fields to the `Token` struct would be considered a breaking change - I doubt there's anyone initializing the `Token` struct directly (and not through the `exchange` method) and if that holds, it shouldn't conflict with existing implementations. or am I missing something?